hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -475,12 +475,12 @@ private void writeToBuffer(HoodieRecord<T> record) {
     }
     Option<IndexedRecord> indexedRecord = getIndexedRecord(record);
     if (indexedRecord.isPresent()) {
-      // Skip the Ignore Record.
+      // Skip the ignored record.
       if (!indexedRecord.get().equals(IGNORE_RECORD)) {
         recordList.add(indexedRecord.get());
       }
     } else {
-      keysToDelete.add(record.getKey());
+      keysToDelete.add(DeleteKey.create(record.getKey(), record.getData().getOrderingVal()));
     }
```

Comment:
Hello, @vinothchandar , can you take a look if you have time ? The only concern is that the new encoding/decoding breaks the compatibility. I have no good idea how to be compatible yet.

But i want to address that before the path, the handle may cause data lost. Comparing to compatibility, correctness is more important i think.

Reply:
Got you, thanks.

Reply:
>I have no good idea how to be compatible yet.

How to Resolve Data Compatibility

Reply:
Have added a new version